### PR TITLE
Disallow `text-transform: capitalize`

### LINKED ISSUE #210: Disallow `text-transform: capitalize;`
I would like to propose disallowing `text-transform: capitalize;`. What this rule does is to Capitalize Every Word For The Affected Element, and while this is borderline OK for English (it still looks pretty bad in my opinion), it leads to incorrect output in many languages (including my own Norwegian).

The capitalization or non-capitalization of strings should be left up to the translators on Translatewiki.net who know the rules and conventions of their language; if you have a message that is lowercase or partly-lowercase, and you for some reason it to having different casing, you should create a new message with different casing.

I discussed this with @jdlrobson who supports adding a rule to disallow this.

### PR DESCRIPTION
**Pull request by jhsoby**
This rule invariably leads to bad/incorrect results in many non-English languages. See rationale in linked issue.

This fixes #210

### CODE REVIEW

**Review by edg2s — CHANGES_REQUESTED** [review]
I'm in two minds whether this is a general rule for our styleguide, or for the /mediawiki subset. I guess for wikimedia projects in general it's a good rule, but maybe not for people who just want to use the wikimedia style. Let's leave it here for now.

Please add a failing test case to default/invalid.css (next to the border/outline test cases).

**Comment by jhsoby** [comment]
For anything that's translatable (including non-MediaWiki) using this is a bad idea. If it's *not* translatable, why would one need to transform text with CSS instead of just changing the text directly?

**Comment by edg2s** [comment]
> For anything that's translatable (including non-MediaWiki) using this is a bad idea. If it's _not_ translatable, why would one need to transform text with CSS instead of just changing the text directly?

Well that's an argument against the rule entirely. Clearly people in monolingual contexts do find a use for it, hence its existence. The question then becomes if the basic config-wikimedia is just stylistic, or opinionated about how you build software.

File: test/fixtures/default/invalid.css [hunk 1]
```diff
@@ -276,4 +278,4 @@ unknown {
 .last {
 	color: #fff;
 	/* stylelint-disable-next-line no-missing-end-of-source-newline */
-}
\ No newline at end of file
+}
```
Comment (edg2s) [inline test/fixtures/default/invalid.css:281]: Your IDE has inserted a newline at the end here. It needs to not exist to assert the linter is working.

Comment (jhsoby) [reply test/fixtures/default/invalid.css:281]: Heh, that's what I get for using GitHub's built-in editing tools. Will have to download it locally to fix this. 🙈

**Comment by jhsoby** [comment]
> > For anything that's translatable (including non-MediaWiki) using this is a bad idea. If it's _not_ translatable, why would one need to transform text with CSS instead of just changing the text directly?
> 
> Well that's an argument against the rule entirely. Clearly people in monolingual contexts do find a use for it, hence its existence. The question then becomes if the basic config-wikimedia is just stylistic, or opinionated about how you build software.

Yes, I do think it's an unnecessary rule to begin with. :-)
However, the point I'm trying to make is that for an ecosystem that is as multilingual as Wikimedia's, it's not just an *unnecessary* rule, but actually a *harmful* one, since it leads to correctly translated text appearing incorrectly. Whenever I see stuff like [this](https://phabricator.wikimedia.org/F37095292) in Norwegian, I automatically assume it was written by someone who doesn't have a grasp of even the basics of Norwegian spelling rules, and thus I immediately lose a lot of trust in that website/publication. And that's something we ought to avoid IMO.

**Comment by Volker-E** [comment]
I agree. In any multilingual context this rule seems possibly harmful. In a specific language context it could be still allowed via override, but you should inform yourself before doing so.

**Comment by jdforrester** [comment]
Yeah, I think we should in general ban `text-transform`, not just `text-transform: capitalize`. It's not safe to apply to any general Wikimedia content.

**Review by jdlrobson — APPROVED** [review]

**Review by jdforrester — DISMISSED** [review]
Per my comment.

**Comment by jdlrobson** [comment]
> Yeah, I think we should in general ban `text-transform`, not just `text-transform: capitalize`. It's not safe to apply to any general Wikimedia content.

I'd rather we did this in a follow up as this is more debatable e.g. usage for acronyms) and we have a LOT of these rules in our existing codebases.

I'd rather not delay something we know is very problematic for something which is more questionable.

**Review by edg2s — APPROVED** [review]

**Comment by jdforrester** [comment]
> > Yeah, I think we should in general ban `text-transform`, not just `text-transform: capitalize`. It's not safe to apply to any general Wikimedia content.
> 
> I'd rather we did this in a follow up issue as this is more debatable (e.g. usage for acronyms) and we have a LOT of these rules in our existing codebases.
> 
> I'd rather not delay something we know is very problematic for something which is more questionable.

The bot auto-disables the rule every time it fails, so your claim that this would be work work seems odd?

**Pull request merged**